tests/test_requests.py

```diff
@@ -1452,3 +1453,17 @@ def test_vendor_aliases():
 
     with pytest.raises(ImportError):
         from requests.packages import webbrowser
+
+def test_chunked_upload():
+    """can safely send generators"""
+    block_server = threading.Event()
+    server = Server.basic_response_server(wait_to_close_event=block_server)
+    data = (i for i in [b'a', b'b', b'c']) 
+
+    with server as (host, port):
+        url = 'http://{0}:{1}/'.format(host, port)
+        r = requests.post(url, data=data, stream=True)
+        block_server.set() # release server block
+
+    assert r.status_code == 200
+    assert r.request.headers['Transfer-Encoding'] == 'chunked'
```

Comment:
So, I'm wondering if we should add a new test file for this kind of test. Urllib3 has a `test_socketlevel.py` file, we might want to consider having a `test_lowlevel.py` with this as the first test inside it.